src/librustc_trans/collector.rs

```diff
@@ -433,6 +434,40 @@ fn check_recursion_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     (instance.def, recursion_depth)
 }
 
+fn check_type_length_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
This seems good; I'm wondering if this is the only place we have to check it. I was originally considering checking this directly in the type interner (i.e., we just couldn't *create* a type that is too big). What do you think about that?